### ISSUE TITLE
Enable fork-pull workflows

### ISSUE THREAD

**Issue by rajsite** [ni/nimble#634]
## 🧹 Tech Debt

Currently nimble does not support fork-pull workflows which makes it difficult for contributions outside of the nimble team. We only support branch-pull and all contributors need to be added to the [nimble GitHub team](https://github.com/orgs/ni/teams/nimble/members) in order to create PRs.

Our beachball integration is currently normal enough that we should be able to switch to a fork-pull workflow. One thing to consider is that with fork-pull secrets are not included which is fine for publishing (github pages, npm, nuget) but may be annoying to lose our other analysis tools (lighthouse, chromatic).

We should switch to fork-pull, particularly to allow for outside contributors when open sourcing #290, but this does not need to be blocked on open sourcing and could be useful now, see #633.

**Comment by rajsite** [comment]
Current status:

Updated workflows to ignore steps that require secrets in: https://github.com/ni/nimble/commit/2b0798a759ddc135ff48d7affd5c2a2c57550a2b
   - The goal of the change was to skip steps if a secret was not defined. You [cannot check the value of secrets in the if conditional](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow) and the docs show creating env variables to hold the secrets but instead I went for making env variables that just hold if the secret was defined. Seemed unnecessary to make the secrets available in the env for every step.

So the workflow for a user that is __not__ on the nimble team is like the following:
1. Create a fork of nimble
2. Optional, but recommended, in the fork go to the actions tab and enable actions so your branches run tests
3. Push you changes to main or a branch in your fork
   - If pushing to main: the workflow will run and try to publish the github pages site in your fork. It will fail unless you enable GitHub pages in your fork.
      - [ ] Maybe we should encourage enabling github pages in forks? Maybe we should make that step ignore failures? Maybe that step should not run in forks? The bulk of the benefit is seeing the Angular and Blazor built app and that can happen in PR as part of the chromatic build, see below.
   - If pushing to a branch: the workflow will run to success but you don't have any secrets so you don't get chromatic or lighthouse reports
      - [ ] Maybe we want to [intentionally leak our chromatic project key](https://www.chromatic.com/docs/github-actions/#forked-repositories)? That wouldn't only be handy for running chromatic in the user fork. Not necessary for the the PR, see below.
   - In either scenario change files are computed against the main of the fork and not against upstream ni/nimble. Beachball did add [code to detect the remote repository](https://github.com/microsoft/beachball/issues/460#issuecomment-1461192637) but it looks like the implementation [checks against the remotes configured in git](https://github.com/microsoft/workspace-tools/pull/34/files#diff-3b1c3a5a0d0ca1ef08b3d3cdb46c97a9efa983ea28348a3454b3cbddd97eb2fdR91) and as far as I can tell the github actions checkout step only configures the `origin` remote to be the fork and not an `upstream` remote to be ni/nimble.
      -  [ ] I think we need to add a step in the workflow to configure nimble as an upstream, i.e. `git remote add upstream <ni/nimble repo url>` (that should be safe to always) or we disable beachball change detection from builds in the fork (make them deal with it only in PR, and even then nimble owners can push the change directly to the branch generally)
4. Create a pull request to nimble with the changes
   - When the PR is opened, by default the workflow won't run for a user that is not part of the nimble team. Someone from nimble needs to approve the workflow run for each commit:
   ![image](https://github.com/ni/nimble/assets/1588923/1af0394c-7c6d-4871-826b-777360576edc)
   - Secrets are not exposed to workflows triggered from pull_request by users not in the nimble team so we lose chromatic and lighthouse status checks. As the chromatic status check is required to merge any PR that means the PR cannot be merged without admin override.
      ![image](https://github.com/ni/nimble/assets/1588923/c6c91118-737d-4bac-b8a1-740f91ac7178)
      - [ ] The approach to enable chromatic and lighthouse checks from PRs without leaking credentials is to do the [`pull_request` + save artifacts + `workflow_run` workflow](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/)

**Comment by jattasNI** [comment]
This issue is currently deprioritized so for now if an external contributor creates a PR then a maintainer will need to copy their changes into a new branch and create a new PR in order for it to be submitted to the repo.

We should capture this in CONTRIBUTING.md and consider closing this issue.
